classes/controller/DummyController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
+
+class DummyControllerCore extends Controller
```

Comment:
This dummy controller (naming can be improved) is necessary as some legacy classes expect `$context->controller` to be not null

Reply:
At first I thought it was used for testing so I was about to suggest moving this into the tests folder But then I saw you use it as a default value when no specific controller is specified

So I understand why you need this, however I find it a bit dangerous to expose a useless controller class like this (because it will be accessible just by its presence)

An alternative could be to use bracket namespace inside your ContextLoadHelper.php file (which @PierreRambaud reminded me of recently), something like this:

```php
namespace {
   class DummyController  extends Controller {
       // ...
   }
}
```

The advantage, if I'm not mistaken, is that with this the DummyController is only accessible by your `ContextLoadHelper` class. To be checked though maybe importing the `ContextLoadHelper` would make it visible but at least it reduces the risk.

Also note that I dropped the `Core` suffix because I don't think this controller should be overridden by anyone.

By the way the dynamic class that you used for test would also do the job (that's what gave me the idea), it's just that it would make the code not very readable.

Reply:
If the previous mentioned solutions are not acceptable for you, at the very least we should heavily document this controller to explain that it's not meant to be used and is only here for internal purposes related with `ContextLoadHelper`

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/21125/commits/8129ee72c6ccbf4e6cee5b7122f9450491c71af3 I used bracket namespace it works good 👍